types/comtac/Cluey/uplink.js

```diff
@@ -0,0 +1,1762 @@
+              break;
+            case 2:
+              string = "falling";
+              break;
+            case 3:
+              string = "both";
+              break;
+            default:
+              string = "notAllowed";
+              break;
+          }
+          obj.payload.data.parameters[
+            obj.payload.data.parameters.length - 1
+          ].values.push(string);
+        }
+        break;
+      default:
+        break;
+    }
+  } while (pointer < data.length - 1);
+
+  return {
+    data: obj,
+    warnings,
+  };
+}
+
+function decodeInfoPayload(data) {
+  const obj = {};
+  const warnings = [];
+
+  obj.port = INFO_PORT;
+  obj.portFunction = "INFO";
+  obj.payloadLength = data.length;
+  obj.payload = {
+    device: deviceHeader,
+  };
+
+  obj.decoder = {
+    version: PAYLOAD_DECODER_VERSION,
+    info: PAYLOAD_DECODER_INFO,
+  };
+
+  return {
+    data: obj,
+    warnings,
+  };
+}
+
+function decodeGpsPayload(data) {
+  const obj = {};
+  const warnings = [];
+
+  obj.port = GPS_PORT;
+  obj.portFunction = "GPS";
+  obj.payloadLength = data.length;
+  obj.payload = {
+    device: deviceHeader,
+    data: {
+      coordinates: {
+        latitude: 0,
+        longitude: 0,
+      },
+    },
+  };
+
+  obj.decoder = {
+    version: PAYLOAD_DECODER_VERSION,
+    info: PAYLOAD_DECODER_INFO,
+  };
+
+  obj.payload.data.coordinates.latitude = parseInt(
+    data.slice(pointer, pointer + 4).toString("hex"),
+    16,
+  );
+  obj.payload.data.coordinates.longitude = parseInt(
+    data.slice(pointer + 5, pointer + 9).toString("hex"),
+    16,
+  );
+
+  return {
+    data: obj,
+    warnings,
+  };
+}
+
+function decodePingPayload(data) {
+  const obj = {};
+  const warnings = [];
+
+  obj.port = PING_PORT;
+  obj.portFunction = "PING";
+  obj.payloadLength = data.length;
+  obj.payload = {
+    device: deviceHeader,
+  };
+
+  obj.decoder = {
+    version: PAYLOAD_DECODER_VERSION,
+    info: PAYLOAD_DECODER_INFO,
+  };
+
+  return {
+    data: obj,
+    warnings,
+  };
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const { port } = event.data;
+  const bytes = Hex.hexToBytes(payload);
+  const decoded = decodeUplink({
+    fPort: port,
+    bytes,
+  });
+
+  const { device } = decoded.data.payload;
+  const { data } = decoded.data.payload;
+  const { warnings } = decoded;
+
+  // Digital inputs
+  for (let i = 0; i < 8; i++) {
+    const digitalInputs = {};
+    const digital = data.digitalInputs[i];
+
+    digitalInputs.inputNr = digital.info.id;
+    digitalInputs.limit = digital.cot.limit;
+    digitalInputs.event = digital.cot.event;
+    digitalInputs.interrogation = digital.cot.interrogation;
+    digitalInputs.cyclic = digital.cot.cyclic;
+
+    digitalInputs.blocked = digital.status.blocked;
+    digitalInputs.state = digital.status.state;
+
+    const timestamp = new Date(digital.timestamp.string);
+
+    emit("sample", {
+      data: digitalInputs,
+      topic: "digital",
```

Comment:
does it make sense to have a separate topic per digital input?

Reply:
As this sensor is a bridge, the topics and sorting may be need changed anyways by the client. So i think its alright for the moment.

Reply:
changing topic in the future is always cumbersome for people using the sensor